src/main/java/com/guciowons/shoppingbasket/Product/ProductService.java

```diff
@@ -6,13 +6,14 @@
 
 @Service
 public class ProductService {
-    private final ProductDao productDao;
 
-    public ProductService(ProductDao productDao) {
-        this.productDao = productDao;
+    private final ProductClient productClient;
+
+    public ProductService(ProductClient productClient) {
+        this.productClient = productClient;
     }
 
     public List<Product> getProducts() {
-        return productDao.getAll();
+        return productClient.getProducts();
```

Comment:
Ogólnie spoko, ale nie widzę tutaj nigdzie obsługi błędów. Jest zaprogramowany tylko tak zwany happy path gdzie wszystko działa. Co się stanie jak ten serwis będzie niedostępny? Co wtedy pokazać jako odpowiedź? Bo aktualnie po prostu poleci `500` jako internal server error co nie jest zbyt ładnym komunikatem dla odbiorcy.

Spróbuj jakoś połapać te wyjątki i zwrócić dla usera coś wartościowego jeśli się uda. Wystarczy coś w stylu `Unable to fetch products from external provider`.